hack/cel-validation/gateway_test.go

```diff
@@ -446,14 +482,21 @@ func TestValidateGateway(t *testing.T) {
 
 	for _, tc := range testCases {
 		t.Run(tc.desc, func(t *testing.T) {
-			gwc := baseGateway.DeepCopy()
-			gwc.Name = fmt.Sprintf("foo-%v", time.Now().UnixNano())
+			gw := baseGateway.DeepCopy()
+			gw.Name = fmt.Sprintf("foo-%v", time.Now().UnixNano())
+
+			if tc.mutate != nil {
+				tc.mutate(gw)
+			}
+			err := k8sClient.Create(ctx, gw)
 
-			tc.mutate(gwc)
-			err := k8sClient.Create(ctx, gwc)
+			if tc.mutateStatus != nil {
+				tc.mutateStatus(gw)
+				err = k8sClient.Status().Update(ctx, gw)
+			}
 
 			if (len(tc.wantErrors) != 0) != (err != nil) {
-				t.Fatalf("Unexpected error while creating Gateway; got err=\n%v\n;want error=%v", err, tc.wantErrors != nil)
+				t.Fatalf("Expected error while creating Gateway but didn't get one; got err=\n%v\n;want error=%v", err, tc.wantErrors != nil)
```

Comment:
Sorry it seems like the prior statement was also not correct. Would:
`Unexpected response while creating Gateway" be more correct?

Because there are tests that don't necessarily "expect an error"

Reply:
Updated, thanks!